File: ruby/sinatra_for_dummies/startup_web_server.rb
require 'rubygems'
require 'sinatra'

class StartupWebServer

  def answer(q)
    if q =~ /the sum of (\d+) and (\d+)/
      return ($1.to_i + $2.to_i).to_s
    elsif q =~ /what's santa clause's real name/
      return "Kris Kringle"
    else
      return "team name"
    end
  end

end

server = StartupWebServer.new

configure do
  set :port, 1337
end

get '/' do
  puts "A request has arrived"
  server.answer(params[:q])
end

File: kotlin/ktor/src/main/kotlin/StartupWebServer.kt
import io.ktor.application.*
import io.ktor.http.*
import io.ktor.response.*
import io.ktor.routing.*
import io.ktor.server.engine.*
import io.ktor.server.netty.*

fun main() {
    StartupWebServer().start()
}

class StartupWebServer(val port: Int = 8080) {
    private var server: ApplicationEngine? = null

    fun start(wait: Boolean = true) {
        server = embeddedServer(Netty, port) {
            routing {
                get("/") {
                    log.info("A request has arrived")
                    val question = call.request.queryParameters["q"]
                    val answer =
                        if (question == null) "team name" else {
                            if (question.matches(Regex(".*what is the sum of (\\d+) and (\\d+)"))) {
                                val (a,b) = Regex(".*what is the sum of (\\d+) and (\\d+)").matchEntire(question)!!.destructured
                                (a.toInt() + b.toInt()).toString()
                            } else "team name"
                        }

                    call.respondText(answer, ContentType.Text.Html)
                }
            }
        }.start(wait)
    }

    fun stop() {
        server?.stop(100,1000)
        server = null
    }
}